src/main/java/com/jakub/todoSandbox/repository/JOOQRepository.java

```diff
@@ -0,0 +1,224 @@
+            var addStep = context.insertInto(STEP)
+                    .set(STEP.NAME, step.name())
+                    .set(STEP.DESCRIPTION, step.description())
+                    .set(STEP.TODO_ID, todoId)
+                    .execute();
+        }
+    }
+
+    @Transactional
+    @Override
+    public void updateStep(long todoId, Step updatedStep) {
+        System.out.println("updateStep() method called with todoId: " + todoId);
+
+        Todo todoToUpdate = findTodoById(todoId)
+                .orElseThrow(() -> new ValidationException("No Todo has been found for todoId: {" + todoId + "}"));
+
+        if (todoToUpdate.steps().stream().anyMatch(step -> Objects.equals(step.id(), updatedStep.id()))) {
+            var updatedRows = context.update(STEP)
+                    .set(STEP.NAME, updatedStep.name())
+                    .set(STEP.DESCRIPTION, updatedStep.description())
+                    .where(STEP.ID.eq(updatedStep.id()))
+                    .and(STEP.TODO_ID.eq(todoId))
+                    .execute();
+
+            if (updatedRows > 0) {
+                System.out.println("Step with stepId: {" + updatedStep.id() + "} has been updated");
+            } else {
+                throw new ValidationException("Failed to update the step with stepId: {" + updatedStep.id() + "}");
+            }
+        } else {
+            throw new ValidationException("The step to update was not found in the todo with todoId: {" + todoId + "}");
+        }
+    }
+
+    @Transactional
+    @Override
+    public void deleteSteps(long todoId, List<Long> stepIds) {
+        System.out.println("deleteSteps() method called with todoId: {" + todoId + "}.");
+
+        Todo doesTodoExist = findTodoById(todoId)
+                .orElseThrow(() -> new ValidationException("No Todo has been found for todoId: {" + todoId + "}."));
+
+        if (doesTodoExist != null) {
+            for (long stepId : stepIds) {
+                boolean stepExists = doesTodoExist.steps().stream()
+                        .anyMatch(step -> Objects.equals(step.id(), stepId));
+
+                if (stepExists) {
+                    var deletedRows = context.deleteFrom(STEP)
+                            .where(STEP.ID.eq(stepId))
+                            .and(STEP.TODO_ID.eq(todoId))
+                            .execute();
+
+                    if (deletedRows > 0) {
+                        System.out.println("Step with stepId: {" + stepId + "} was removed.");
+                    } else {
+                        throw new ValidationException("Failed to delete the step with stepId: {" + stepId + "}.");
+                    }
+                } else {
+                    throw new ValidationException("The step with stepId: {" + stepId + "} does not exist in the todo with todoId: {" + todoId + "}.");
+                }
+            }
+        } else {
+            throw new ValidationException("Todo with the provided id does not exist.");
+        }
```

Comment:
Why do we need to check if all these steps exist? Can't we just delete them without checking? What's the disadvantage of doing that?